prereise/gather/griddata/transmission/geometry.py

```diff
@@ -0,0 +1,286 @@
+        conductors if ``circuits`` > 1. Units are meters.
+    :param tuple c: the (x, y) location of the single 'C' phase conductor if
+        ``circuits`` == 1, or the ((x1, y1), (x2, y2), ...) locations of the 'C' phase
+        conductors if ``circuits`` > 1. Units are meters.
+    :param int circuits: the number of circuits on the tower.
+    """
+
+    a: tuple
+    b: tuple
+    c: tuple
+    circuits: int = 1
+    equivalent_distance: float = field(init=False)
+    equivalent_height: float = field(init=False)
+    phase_self_distances: dict = field(init=False)
+    equivalent_reflected_distance: float = field(init=False)
+
+    def __post_init__(self):
+        self.validate_input_types()  # defined in DataclassWithValidation
+        if not (len(self.a) == len(self.b) == len(self.c)):
+            raise ValueError("each phase location must have the same length")
+        if self.circuits == 1 and len(self.a) == 2 and isinstance(self.a[0], float):
+            # Single-circuit specified as (x, y) will be converted to ((x, y))
+            self.a = (self.a,)
+            self.b = (self.b,)
+            self.c = (self.c,)
+        self.calculate_distances()
+
+    def calculate_distances(self):
+        self.true_distance = {
+            "ab": _geometric_mean_euclidian(self.a, self.b),
+            "ac": _geometric_mean_euclidian(self.a, self.c),
+            "bc": _geometric_mean_euclidian(self.b, self.c),
+        }
+        # 'Equivalent' distances are geometric means
+        self.equivalent_distance = geometric_mean(self.true_distance.values())
+        self.equivalent_height = geometric_mean(
+            [self.a[0][1], self.b[0][1], self.c[0][1]]
+        )
+        if self.circuits == 1:
+            self.calculate_single_circuit_distances()
+        else:
+            self.calculate_multi_circuit_distances()
+
+    def calculate_single_circuit_distances(self):
+        # The distance bounced off the ground, or 'reflected', is used for
+        # single-circuit capacitance calculations
+        self.reflected_distance = {
+            "ab": _euclidian(self.a[0], (self.b[0][0], -self.b[0][1])),  # a -> b'
+            "ac": _euclidian(self.a[0], (self.c[0][0], -self.c[0][1])),  # a -> c'
+            "bc": _euclidian(self.b[0], (self.c[0][0], -self.c[0][1])),  # b -> c'
+        }
+        self.equivalent_reflected_distance = geometric_mean(
+            self.reflected_distance.values()
+        )
+        # Single circuit, so we don't have distances between bundles of the same phase
+        self.phase_self_distances = None
+
+    def calculate_multi_circuit_distances(self):
+        self.phase_self_distances = [
+            geometric_mean(
+                [
+                    _euclidian(p0, p1)
+                    for i, p0 in enumerate(phase)
+                    for j, p1 in enumerate(phase)
+                    if i > j
+                ]
+            )
+            for phase in (self.a, self.b, self.c)
```

Comment:
Not sure whether this is equivalent to:
```python
geometric_mean( _euclidian(p0, p1) for p0, p1 in combinations(phase, 2) for phase in (self.a, self.b, self.c)
```

Reply:
Nice! Yeah `combinations` is definitely the cleaner way to do this. Done.